great_expectations/profiler/parameter_builder/multi_batch_bootstrapped_metric_distribution_parameter_builder.py

```diff
@@ -1,60 +1,77 @@
+from great_expectations.profiler.parameter_builder.multi_batch_parameter_builder import (
+    MultiBatchParameterBuilder,
+)
+from great_expectations.profiler.parameter_builder.parameter import Parameter
+from great_expectations.profiler.profiler_rule.rule_state import RuleState
+from great_expectations.validator.validation_graph import MetricConfiguration
+from great_expectations.validator.validator import Validator
 
 
 class MultiBatchBootstrappedMetricDistributionParameterBuilder(
     MultiBatchParameterBuilder
 ):
     """
+    Defines the MultiBatchBootstrappedMetricDistributionParameterBuilder.
+
     Builds parameters from the p_values of the distribution of a metric observed from a set of batches identified in the
     batch_ids.
     """
 
     def __init__(
         self,
         *,
-        parameter_id,
-        batch_request,
-        metric_name,
-        metric_value_kwargs,
-        p_values,
-        data_context
+        parameter_name: str,
+        batch_request: BatchRequest,
+        metric_name: str,
+        metric_value_kwargs: Union[str, dict],
+        p_values: List[float],
+        data_context: Optional[DataContext] = None
     ):
         """
         Create a MultiBatchBootstrappedMetricDistributionParameterBuilder.
 
         The ParameterBuilder will build parameters for the active domain from the rule.
 
         Args:
-            parameter_id: the id for this ParameterBuilder
+            parameter_name: the name of the parameter handled by this ParameterBuilder
             batch_request: BatchRequest elements that should be used to obtain additional batch_ids
             metric_name: metric from which to build the parameters
             metric_value_kwargs: value kwargs for the metric to be built
             p_values: the p_values for which to return metric value estimates
         """
         super().__init__(
-            parameter_id=parameter_id,
-            data_context=data_context,
+            parameter_name=parameter_name,
             batch_request=batch_request,
+            data_context=data_context,
         )
+
         self._metric_name = metric_name
         self._metric_value_kwargs = metric_value_kwargs
         self._p_values = p_values
 
     # TODO: <Alex>ALEX -- There is nothing about "p_values" in this implementation; moreover, "p_values" would apply only to certain values of the "metric_name" -- this needs to be elaborated.</Alex>
-    def _build_parameters(self, *, rule_state, validator, batch_ids, **kwargs):
+    def _build_parameters(
+        self,
+        *,
+        rule_state: Optional[RuleState] = None,
+        validator: Optional[Validator] = None,
+        batch_ids: Optional[List[str]] = None,
+        **kwargs
+    ) -> Parameter:
         samples = []
         for batch_id in batch_ids:
-            metric_domain_kwargs = copy(rule_state.active_domain.domain_kwargs)
+            metric_domain_kwargs = copy(rule_state.active_domain["domain_kwargs"])
```

Comment:
Just a general question and not critical to answer before merging this PR: Could we use DictDot for domain_kwargs so that we can access using the dot notation? Would this help us at all (in this case and any others with dicts) in terms of easier refactor later if we change the name over using dict["string"] accessors?

Reply:
I like the idea; I think this should be approached in a "wholesale" manner as single-purpose-focused PR.  Happy to discuss.

Reply:
This is now `https://superconductive.atlassian.net/browse/GE-16`